server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointSnapshotTransferManagerWithoutMetadata.java

```diff
@@ -0,0 +1,98 @@
+import org.opensearch.index.translog.transfer.FileSnapshot.TranslogFileSnapshot;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Translog transfer manager to transfer {@link TranslogCheckpointSnapshot} by transfering {@link TranslogFileSnapshot} and {@link CheckpointFileSnapshot} separately
+ *
+ * @opensearch.internal
+ */
+public class TranslogCheckpointSnapshotTransferManagerWithoutMetadata implements TranslogCheckpointSnapshotTransferManager {
+
+    private final TransferService transferService;
+    private final FileTransferTracker fileTransferTracker;
+
+    public TranslogCheckpointSnapshotTransferManagerWithoutMetadata(
+        TransferService transferService,
+        FileTransferTracker fileTransferTracker
+    ) {
+        this.transferService = transferService;
+        this.fileTransferTracker = fileTransferTracker;
+    }
+
+    @Override
+    public void transferTranslogCheckpointSnapshot(
+        Set<TranslogCheckpointSnapshot> toUpload,
+        Map<Long, BlobPath> blobPathMap,
+        LatchedActionListener<TranslogCheckpointSnapshot> latchedActionListener,
+        WritePriority writePriority
+    ) throws Exception {
+        for (TranslogCheckpointSnapshot tlogAndCkpTransferFileSnapshot : toUpload) {
+            Set<TransferFileSnapshot> filesToUpload = new HashSet<>(2);
+            Set<Exception> exceptionList = new HashSet<>(2);
+
+            String tlogFileName = tlogAndCkpTransferFileSnapshot.getTranslogFileName();
+            String ckpFileName = tlogAndCkpTransferFileSnapshot.getCheckpointFileName();
+
+            if (!fileTransferTracker.uploaded(tlogFileName)) {
+                filesToUpload.add(tlogAndCkpTransferFileSnapshot.getTranslogFileSnapshot());
+            }
+            if (!fileTransferTracker.uploaded(ckpFileName)) {
+                filesToUpload.add(tlogAndCkpTransferFileSnapshot.getCheckpointFileSnapshot());
+            }
+
+            assert !filesToUpload.isEmpty();
+
+            AtomicBoolean listenerProcessed = new AtomicBoolean();
+            final CountDownLatch latch = new CountDownLatch(filesToUpload.size());
+
+            Set<TransferFileSnapshot> successFiles = new HashSet<>();
+            Set<TransferFileSnapshot> failedFiles = new HashSet<>();
+
+            LatchedActionListener<TransferFileSnapshot> fileUploadListener = new LatchedActionListener<>(
+                ActionListener.wrap(successFiles::add, exceptionList::add),
+                latch
+            );
+            ActionListener<TransferFileSnapshot> actionListener = ActionListener.runAfter(fileUploadListener, () -> {
+                if (latch.getCount() == 0 && listenerProcessed.compareAndSet(false, true)) {
+                    if (exceptionList.isEmpty()) {
+                        latchedActionListener.onResponse(tlogAndCkpTransferFileSnapshot);
```

Comment:
lets add an assertion here that succeededFiles count is equal to filesToUpload size and failedFiles count is 0.

Reply:
If we can gave deterministic check that would be even better.

Reply:
as we are already checking `exceptionList.empty()`, we don't need to check for failedFiles count. adding assertion 
`assert succeededFiles.size() == filesToUpload.size();`